plugins/logging/pkg/logging/admin.go

```diff
@@ -370,35 +411,115 @@ func (p *Plugin) convertProtobufToDashboards(
 	}
 }
 
+func (p *Plugin) setOpensearchClient() {
+	expBackoff := backoff.Exponential(
+		backoff.WithMaxRetries(0),
+		backoff.WithMinInterval(5*time.Second),
+		backoff.WithMaxInterval(1*time.Minute),
+		backoff.WithMultiplier(1.1),
+	)
+	b := expBackoff.Start(p.ctx)
+
+	cluster := &opsterv1.OpenSearchCluster{}
+
+FETCH:
+	for {
+		select {
+		case <-b.Done():
+			p.logger.Warn("plugin context cancelled before Opensearch object created")
+		case <-b.Next():
+			err := p.k8sClient.Get(p.ctx, types.NamespacedName{
+				Name:      p.opensearchCluster.Name,
+				Namespace: p.storageNamespace,
+			}, cluster)
+			if err != nil {
+				if k8serrors.IsNotFound(err) {
+					p.logger.Info("waiting for k8s object")
+					continue
+				}
+				p.logger.Errorf("failed to check k8s object: %v", err)
+				continue
+			}
+			break FETCH
+		}
+	}
+
+	username, password, err := helpers.UsernameAndPassword(p.ctx, p.k8sClient, cluster)
+	if err != nil {
+		p.logger.Errorf("failed to get cluster details: %v", err)
+		return
+	}
+
+	// Set sane transport timeouts
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.DialContext = (&net.Dialer{
+		Timeout: 5 * time.Second,
+	}).DialContext
+	transport.TLSHandshakeTimeout = 5 * time.Second
+	transport.TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
Would it be doable to obtain the opensearch cacert here?

Reply:
Possibly, but its internal comms with a selfsigned cert so I was les concerned about it.